### DISCUSSION TITLE
I need help deciding on adding Livewire support

### DISCUSSION THREAD

**Discussion by mkocansey** [bladewindui/ui#185] [Polls]
There have been quite a couple of requests for BladewindUI to include support for Livewire. I agree. Especially since Livewire is gaining popularity & usage and will need beautiful components like we have in BladewindUI.

Currently BladewindUI intentionally uses **_vanilla Javascript_** for all frontend interactions (except in the Datepicker).

My dilemma. Considering not everyone uses Livewire in their projects, will it be better to add Livewire support to the current  BladewindUI library or will it be better to simply have a new library with all the components but with support for Livewire?

**Comment by BeBel42** [comment]
Would creating a new library force you to repeat you code changes?
If that means you have to maintain the two libraries separately for each component, I would not see any advantage to create a new library.

**Reply by mkocansey** [reply]
@BeBel42 sadly yes! I will need to maintain two codebases if I create a separate library with Livewire support. Come to think of it, I don't see an advantage either

**Reply by BeBel42** [reply]
I think we're set then!
Having two codebases would mean two separate documentations, two separate repos, code to maintain, etc.
Adding support to the current library would mean supporting only one codebase, and just create another section in the docs.

**Reply by szajens** [reply]
I've thought it through, indeed, it wouldn't make sense to maintain two codebases
